### ISSUE TITLE
Unclear error interface of `SDL_WaitEventTimeout`

### ISSUE THREAD

**Issue by rohlem** [libsdl-org/SDL#5052]
It seems to me there is an interface issue with `SDL_WaitEventTimeout`.
This concern was [brought up (but seemingly overlooked/ignored) in a comment on issue #3976](https://github.com/libsdl-org/SDL/issues/3976#issuecomment-792743499) .

As [documentation](https://wiki.libsdl.org/SDL_WaitEventTimeout) states:
> Returns 1 on success or 0 if there was an error while waiting for events; call SDL_GetError() for more information.
> **This also returns 0 if the timeout elapsed without an event arriving.**

This means a user of this function cannot know whether the timeout elapsed, or an error occured (unless they are lucky enough that checking the time after the function returned ensures the timeout hasn't expired yet).
While it would seem safe to check `SDL_GetError()` in that case, the [corresponding documentation warns against this](https://wiki.libsdl.org/SDL_GetError):
> The message is only applicable **when an SDL function has signaled an error. You must check the return values of SDL function calls** to determine when to appropriately call SDL_GetError().

And so it seems that no call to `SDL_WaitEventTimeout` that returned `0` can be correctly handled, judging by documentation alone.

**Comment by slouken** [comment]
You're right, this is unclear, and we should fix it for SDL 2.1 when we can change the ABI. However looking at the code, it looks like the only time it will return an error is when you call it after the event system has been shut down. For all practical cases, you can treat 0 as timeout elapsed with no error possible.

**Comment by rohlem** [comment]
Thank you for the quick response!
> For all practical cases, you can treat 0 as timeout elapsed with no error possible.

This means the error case is (currently) unnecessary? Should that option be kept open for future 2.0.x releases,
or would it be feasible to "retire" these functions having the option to return errors?
As in, clearly state "this function will not error (if the event system was initialized) after version 2.0.16". (Which would also apply to `SDL_WaitEvent`.)

**Comment by slouken** [comment]
Sure, I think that would be fine to update the documentation in that way.

**Comment by rohlem** [comment]
Just realized that I never submitted the doc change and didn't remember why, so I dug up my notes again.
In case someone else finds this and is curious:

I looked back through old versions up until 2.0.0, and it seemed like the error conditions stayed unchanged.
However, when looking into locking the event queue, I saw that on some platforms (I think Linux)
the implementation used some OS-level mutex functions that were documented as (spuriously?) fallible.
Through that the error condition seemed actually reachable (even if pretty unlikely),
and detailing the error scenario alone seemed unproductive compared to my initial expectation of ruling it out completely.
